nativelib/src/main/scala/scala/scalanative/unsafe/package.scala

```diff
@@ -73,13 +75,13 @@ package object unsafe {
   type CBool = Boolean
 
   /** The C/C++ 'size_t' type. */
```

Comment:
I am wondering if maybe it makes more sense to use `CSize` and `CSSize` if we are going to remove the `Word` and `UWord` indirection. This way APIs that use `size_t` and `ssize_t` can just use the types as is or the C APIs can just do a `type size_t = CSize`. I think Lee and I moved away from that standard and started just using `CSize` after that was added to the API. This also means we are reducing the number of types and it is clear they are C types like the other types that start with `C`.

This also means that any explicit conversions would become `toCSize` and `toCSSize`. I think since there is a cognitive load needed by the developer for C interaction, anything we can do to map closely helps.

Reply:
My feel on this is that we want `Size`/`USize` to be part of the regular set of Scala integer types rather than specifically for C FFI, since just like unsigned types Scala Native lets us efficiently represent platform-dependent types. Even though `CSize` is just an alias, I like having it as a separate name so that developers are sure to always use the FFI-supported types when defining extern functions.

Reply:
Ah yeah, that's a fair point. I didn't include `i` for the signed type since Scala doesn't have prefixes for the built-in signed types but renaming could work. We _do_ need new types though, since you can't safely interact with `size_t` APIs with just an `Int` or a `Long` in a cross-platform manner.

Reply:
I finally figured out the `size_t` in the unsigned "size" type and the `ssize_t` is the signed "size" type. Wow.

Reply:
Yup, I just renamed `Word` to `Size` since word has a lot of different definitions when talking about memory.

Yeah, that's unfortunately a bit confusing. Thankfully we have the typechecker to make sure you don't swap the two.